File: app/src/main/java/uz/revolution/mantiq_qani/manager/MyManager.java
package uz.revolution.mantiq_qani.manager;


import uz.revolution.mantiq_qani.model.MyModel;

import java.util.ArrayList;

public class MyManager {
    private ArrayList<MyModel> questions;
    private int level = 0;
    private int totalScore=0;
    private final int deltaScore = 2;
    private final int maxScore = 10;
    private int currentScore=maxScore;



    public void setQuestions(ArrayList<MyModel> questions) {
        this.questions = questions;
    }



    public void setCurrentScore(int currentScore) {
        this.currentScore = currentScore;
    }

    public MyModel getCurrentQuestion() {
        return questions.get(level);
    }


    public int getAnswerLength(){
        return getCurrentQuestion().getAnswer().length();
    }

    private String getAnswer() {
        return getCurrentQuestion().getAnswer();
    }

    public String getQuestion() {
        return getCurrentQuestion().getQuestion();
    }

    public boolean checkAnswer(String userAnswer) {

        if (userAnswer.equalsIgnoreCase(getAnswer().trim().replaceAll(" ", ""))) {

            totalScore = totalScore + currentScore;
            currentScore = maxScore;
            level++;

            return true;
        } else {

            if (currentScore > deltaScore) {
                currentScore = currentScore - deltaScore;
            }
            return false;
        }

    }

    public boolean isEnd() {
        return level>= questions.size();
    }

    public boolean hasNext() {
        return level < questions.size();
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }

    public int getTotalScore() {
        return totalScore;
    }

    public void setTotalScore(int totalScore) {
        this.totalScore = totalScore;
    }

    public int getCurrentScore() {
        return currentScore;
    }
}

File: app/src/main/java/uz/revolution/mantiq_qani/manager/MyManager.kt
package uz.revolution.mantiq_qani.manager

import uz.revolution.mantiq_qani.model.MyModel
import java.util.*

class MyManager {
    private var questions: ArrayList<MyModel>? = null
    var level = 0
    var totalScore = 0
    private val deltaScore = 2
    private val maxScore = 10
    var currentScore = maxScore
    fun setQuestions(questions: ArrayList<MyModel>?) {
        this.questions = questions
    }

    val currentQuestion: MyModel
        get() = questions!![level]

    fun getAnswerLength(): Int {
        return currentQuestion.answer.length
    }

    private fun getAnswer(): String {
        return currentQuestion.answer
    }

    fun getQuestion(): String {
        return currentQuestion.question
    }

    fun checkAnswer(userAnswer: String): Boolean {
        return if (userAnswer.equals(
                getAnswer().trim { it <= ' ' }.replace(" ".toRegex(), ""),
                ignoreCase = true
            )
        ) {
            totalScore += currentScore
            currentScore = maxScore
            level++
            true
        } else {
            if (currentScore > deltaScore) {
                currentScore -= deltaScore
            }
            false
        }
    }

    fun isEnd(): Boolean {
        return level >= questions!!.size
    }

    operator fun hasNext(): Boolean {
        return level < questions!!.size
    }
}